Unix/tools/allocdump

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+SCRIPTDIR=$(dirname "$0")
+BINDIR=/opt/omi/bin
+OMISERVER=${BINDIR}/omiserver
+GDB=gdb
+GDB_SCRIPT="$SCRIPTDIR/gdb_allocdump"
+MINIMUM_VERSION="100401000"   # version 1.4.1-0
+CONFIGEDITOR=${BINDIR}/omiconfigeditor
+CONF_FILE=/etc/opt/omi/conf/omiserver.conf
+
+# Verify that the GDB script file exists
+if [ ! -f ${GDB_SCRIPT} ]; then
+    echo "Gdb script file ${GDB_SCRIPT} does not exist"
+    exit 1
+fi
+
+# Verify that gdb exists
+if [ ! -x "$(command -v ${GDB})" ]; then 
+    echo "${GDB} not found in your PATH"
+    exit 1
+fi
+
+# Verify sudo
+EFFECTIVE_ID="$(id -u)"
+SUDO=
+if [ ${EFFECTIVE_ID} -ne 0 ]; then
+    SUDO=sudo
+    if [ ! -x "$(command -v ${SUDO})" ]; then 
+        echo "${SUDO} not found in your PATH"
+        exit 1
+    fi
+    ${SUDO} -v >&1 >/dev/null 
+    if [ $? -eq 1 ]; then
+        echo "You do not have access to run ${SUDO}"
+        exit 1
+    fi
+fi
+
+# Verify that omiserver process is running
+ps -ef | fgrep ${OMISERVER} | fgrep -v fgrep 2>&1 >/dev/null
+if [ $? -eq 1 ]; then
+    echo "Omiserver is not running"
+    exit 1
+fi
+
+# Verify that omiserver.conf is set up properly
+if [ ! -f ${CONF_FILE} ]; then 
+    echo "${CONF_FILE} does not exist"
+    exit 1
+fi
+if [ ! -x ${CONFIGEDITOR} ]; then 
+    echo "${CONFIGEDITOR} does not exist"
+    exit 1
+fi
+for i in INFO DEBUG VERBOSE; do 
+    ${CONFIGEDITOR} loglevel -q $i < ${CONF_FILE} 2>&1 >/dev/null; 
+    [ $? -eq 0 ] && found=1 && break; 
+done
+if [ "$found" != "1" ]; then
+    echo "Please set loglevel to INFO or higher in ${CONF_FILE}"
+    exit 1
+fi
+
+# Verify that omiserver meets minimum version requirement
+VERSION_LINE="$(${OMISERVER} -v)"
+PRODUCT="$(echo $VERSION_LINE | awk -F" |-" '{print $2}')"
+if [ ${PRODUCT} != "OMI" ]; then
+    echo "Incorrect version information in omiserver -v"
+    exit 1
+fi
+VERSION_TEMP="$(echo $VERSION_LINE | awk -F" |-" '{print $3}')"
+VERSION_BUILD="$(echo $VERSION_LINE | awk -F" |-" '{print $4}')"
+VERSION_MAJOR="$(echo $VERSION_TEMP | awk -F. '{print $1}')"
+VERSION_MINOR="$(echo $VERSION_TEMP | awk -F. '{print $2}')"
+VERSION_PATCH="$(echo $VERSION_TEMP | awk -F. '{print $3}')"
+if [ ${#VERSION_MAJOR} -gt 3 \
+     -o ${#VERSION_MINOR} -gt 3 \
+     -o ${#VERSION_PATCH} -gt 2 \
+     -o ${#VERSION_BUILD} -gt 3 ]; then
+    echo "Invalid version number: ${VERSION_MAJOR}:${VERSION_MINOR}:${VERSION_PATCH}:${VERSION_BUILD}"
```

Comment:
That's not exactly an "invalid version number", it's a version number where a field is too long. Make error message more meaningful please.

Reply:
Changed error message to "Version number exceeds maximum allowed".